stan/math/prim/mat/prob/bernoulli_logit_glm_lpmf.hpp

```diff
@@ -0,0 +1,150 @@
+     * this can also be a single binary value;
+     * @tparam T_x type of the matrix of independent variables (features); this
+     * should be an Eigen::Matrix type whose number of rows should match the 
+     * length of n and whose number of columns should match the length of beta
+     * @tparam T_beta type of the weight vector;
+     * this can also be a single double value;
+     * @tparam T_alpha type of the intercept;
+     * this can either be a vector of doubles of a single double value;
+     * @param n binary vector parameter
+     * @param x design matrix
+     * @param beta weight vector
+     * @param alpha intercept (in log odds)
+     * @return log probability or log sum of probabilities
+     * @throw std::domain_error if theta is infinite.
+     * @throw std::invalid_argument if container sizes mismatch.
+     */
+    template <bool propto, typename T_n, typename T_x, typename T_beta,
+              typename T_alpha>
+    typename return_type<T_x, T_beta, T_alpha>::type
+    bernoulli_logit_glm_lpmf(const T_n &n, const T_x &x, const T_beta &beta,
+                             const T_alpha &alpha) {
+      static const std::string function = "bernoulli_logit_glm_lpmf";
+      typedef typename stan::partials_return_type<T_n, T_x, T_beta,
+                                                  T_alpha>::type
+        T_partials_return;
+
+      using stan::is_constant_struct;
+      using std::exp;
+      using Eigen::Dynamic;
+      using Eigen::Matrix;
+
+      if (!(stan::length(n) && stan::length(x) && stan::length(beta)))
+        return 0.0;
+
+      T_partials_return logp(0.0);
+
+      check_bounded(function, "Vector of dependent variables", n, 0, 1);
+      check_not_nan(function, "Matrix of independent variables", x);
+      check_not_nan(function, "Weight vector", beta);
+      check_not_nan(function, "Intercept", alpha);
+      check_consistent_sizes(function, "Rows in matrix of independent variables",
+                             x.col(0), "Vector of dependent variables",  n);
+      check_consistent_sizes(function, "Columns in matrix of independent variables",
+                             x.row(0), "Weight vector",  beta);
+
+      if (!include_summand<propto, T_x, T_beta, T_alpha>::value)
+        return 0.0;
+
+      const size_t N = x.col(0).size();
+      const size_t M = x.row(0).size();
+      
+      Matrix<double, Dynamic, 1> signs(N,1);
+      {
+        scalar_seq_view<T_n> n_vec(n);
+        for (size_t n = 0; n < N; ++n) {
+          signs[n] = 2 * n_vec[n] - 1;
+        }
+      }
+      Matrix<T_partials_return, Dynamic, 1> beta_dbl(M, 1);
+      {
+        scalar_seq_view<T_beta> beta_vec(beta);
+        for (size_t m = 0; m < M; ++m) {
+          beta_dbl[m] = value_of(beta_vec[m]);
+        }
+      }
+      Matrix<T_partials_return, Dynamic, Dynamic> x_dbl = value_of(x);
+
+      Eigen::Array<T_partials_return, Dynamic, 1> ntheta = signs.array() * 
```

Comment:
operators for continued operations always go on the following line (in both math typesetting and code---makes it easier to scan down the left for structure).